clang/include/clang/Basic/BuiltinsAMDGPU.def

```diff
@@ -69,6 +69,7 @@ BUILTIN(__builtin_amdgcn_iglp_opt, "vIi", "n")
 BUILTIN(__builtin_amdgcn_s_dcache_inv, "v", "n")
 BUILTIN(__builtin_amdgcn_buffer_wbinvl1, "v", "n")
 BUILTIN(__builtin_amdgcn_fence, "vUicC*", "n")
+BUILTIN(__builtin_amdgcn_masked_fence, "vUiUicC*", "n")
```

Comment:
Should put some documentation in LanguageExtensions for the arguments

Reply:
Added docs with tests. I also documented the other fence intrinsic